src/OSPSuite.Core/Domain/EntityFormulaPath.cs

```diff
@@ -0,0 +1,39 @@
+﻿using OSPSuite.Core.Domain.Formulas;
+
+namespace OSPSuite.Core.Domain
+{
+   /// <summary>
+   ///    Simple class to hold an entity (typically an IUsingFormula), the referenced formula and given path entry in the
+   ///    formula
+   ///    <remarks>
+   ///       We are using IEntity instead of IUsingFormula to ensure that the caller is not tempted to do
+   ///       usingFormula.Formula. In most cases, this would be right, except when dealing with RHSFormula in parameters.
+   ///    </remarks>
+   /// </summary>
+   public class EntityFormulaPath
+   {
+      public IEntity Entity { get; }
```

Comment:
See note above. Not using IUsingFormula so that caller always used the formula member if required